js/assessments/wordComplexityAssessment.js

```diff
@@ -90,15 +91,42 @@ var calculateComplexity = function( wordCount, wordComplexity, i18n ) {
  * @returns {Array} A list with markers
  */
 var wordComplexityMarker = function( paper, researcher ) {
-	var wordComplexity = researcher.getResearch( "wordComplexity" );
-	var complexWords = filterComplexity( wordComplexity );
-	return map( complexWords, function( complexWord ) {
-		complexWord.word = removeSentenceTerminators( complexWord.word );
-		complexWord.word = removePunctuation( complexWord.word );
+	var sentences = researcher.getResearch( "wordComplexity" );
+
+	return flatMap( sentences, function( sentence ) {
+		var words = sentence.words;
+		sentence = sentence.sentence;
```

Comment:
`sentence = sentence.sentence`?

Also, this introduces type juggling, which might cause confusion later on.